pkg/graveler/ref/commit_ordered_iterator.go

```diff
@@ -113,14 +89,42 @@ func (i *KVOrderedCommitIterator) Value() *graveler.CommitRecord {
 }
 
 func (i *KVOrderedCommitIterator) Err() error {
-	if i.err == nil {
+	if i.err != nil {
+		return i.err
+	}
+	if i.it != nil {
 		return i.it.Err()
 	}
-	return i.err
+	return nil
 }
 
 func (i *KVOrderedCommitIterator) Close() {
 	if i.it != nil {
 		i.it.Close()
+		i.it = nil
```

Comment:
We basically want that for all iterators, no?

Reply:
The ones that holds internal pointer to implementation - I'll map the rest - I think I went over the pkg/kv in this pr.